projects/mci/src/precharge_control.c

```diff
@@ -17,16 +17,22 @@ static PrechargeControlStorage s_precharge_storage = { 0 };
 PrechargeControlStorage *test_get_storage(void) {
   return &s_precharge_storage;
 }
+
 StatusCode prv_set_precharge_control(PrechargeControlStorage *storage, const GpioState state) {
   gpio_set_state(&storage->precharge_control, state);
-  gpio_set_state(&storage->precharge_control2, state);
   return STATUS_CODE_OK;
 }
 
-void prv_precharge_completed(const GpioAddress *address, void *context) {
+void prv_precharge_monitor(const GpioAddress *address, void *context) {
   PrechargeControlStorage *storage = context;
-  storage->state = MCI_PRECHARGE_CHARGED;
-  CAN_TRANSMIT_PRECHARGE_COMPLETED();
+  if (storage->state == MCI_PRECHARGE_INCONSISTENT) {
```

Comment:
do you think it would be more clear if this was in chronological order? something like
```
if (storage->state == MCI_PRECHARGE_DISCHARGED) {
  ...
else {
  ...
  CAN_TRANSMIT_...
}
```